src/Lifecycle.js

```diff
@@ -436,7 +450,13 @@ export function stopMatrixClient() {
     if (cli) {
         cli.stopClient();
         cli.removeAllListeners();
-        cli.store.deleteAllData();
-        MatrixClientPeg.unset();
     }
+
+    if (clearStores) {
+        // note that we have to do this *after* stopping the client, but
+        // *before* clearing the MatrixClientPeg.
```

Comment:
because the client will complain if you try to clear the store while it's running, and once you clear the matrixclientpeg, you don't have a reference to client so you can tell it to clear the store...